File: modules/BBWM.Maintenance/MaintenanceSettings.cs
using BBWM.SystemSettings;

namespace BBWM.Maintenance;


public enum MaintenanceOptions
{
    Basic = 0,
    External = 1
}

/// <summary>
/// Represents maintenance mode settings.
/// </summary>
public class MaintenanceSettings : IMutableSystemConfigurationSettings
{
    public MaintenanceOptions? Option { get; set; } = MaintenanceOptions.Basic;

    public DateTime? Start { get; set; } = DateTime.Today.AddDays(-1);

    public DateTime? End { get; set; } = DateTime.Today.AddMinutes(-1);

    public string Message { get; set; } = "The site is currently under maintenance. Thank you for your patience.";

    public bool? IsActive { get; set; }

    public string ExternalApiUrl { get; set; }
}

File: project/BBWT.Client/src/app/main/system-configuration/classes/maintenance-settings.ts
import { SettingsSectionsName } from "../settings-sections-name";
import { SystemConfigurationResolveData } from "../system-configuration-resolve-data";

export enum MaintenanceOptions {
    Basic = 0,
    External = 1
}

export class MaintenanceSettings {
    option: MaintenanceOptions;
    start: Date;
    end: Date;
    message: string;
    isActive: boolean;
    externalApiUrl: string;

    static parse(data: SystemConfigurationResolveData) {
        return data.getConfigSection<MaintenanceSettings>(SettingsSectionsName.MaintenanceSettings);
    }
}
